docs/sources/mimir/get-started/about-grafana-mimir-architecture/about-ingest-storage-architecture/index.md

```diff
@@ -0,0 +1,163 @@
+
+After the querier executes the query, it returns the results to the query-frontend for aggregation. The query-frontend then returns the aggregated results to the client.
+
+#### Data freshness on the read path
+
+Ingesters consume from Kafka asynchronously while serving queries. By default, ingesters don't ensure that samples written to Kafka are appended to the in-memory TSDB and on-disk WAL before receiving a query. This allows ingesters to whitstand Kafka outages without rejecting queries. The impact of Kafka outages is that
+queries that ingesters serve may return stale data and don't provide a read-after-write guarantee.
+
+During normal operations, the delay in ingesting data in ingesters should be below 1 second. This is also known as the end-to-end latency of ingestion.
+
+While this latency is usually insignificant for interactive queries, it can challenge applications that need queries to observe all previous writes.
+For example, the Mimir [ruler](../../../references/architecture/components/ruler) needs samples from earlier rules in a rule group to be available when evaluating later rules in the same group.
+
+To preserve the read-after-write guarantee, clients can add the `X-Read-Consistency: strong` HTTP header to queries. When a query includes this header, the query-frontend fetches the latest offsets from Kafka for all active partitions and propagates these offsets to ingesters. Each ingester waits until it has consumed up to the specified offset for its partition before running the query. This ensures the query observes all samples written to Kafka before the query-frontend received the query, providing strong read consistency at the cost of increased query latency.
+
+By default, the ingester waits up to 20 seconds for the record to be consumed before rejecting the strong read consistency query.
+
+## Kafka
+
+Kafka is a central part of ingest storage architecture. It's responsible for replicating and durably persisting incoming data
+and for making this data available to ingesters.
+
+The use of the Kafka protocol is limited in Grafana Mimir. This allows you to use many Kafka-compatible solutions.
+
+### How Mimir uses Kafka
+
+On the write path, the distributor requires the Kafka cluster to be available for push requests to succeed.
+The distributor issues direct Produce API calls to the Kafka brokers and doesn't produce in transactions or need support for idempotent writes.
+
+Both the ingester and the distributor use the Metadata API to discover the leaders for a topic-partition.
+
+The Mimir ingester uses Kafka consumer groups to persist how far an ingester has consumed from its assigned partition. Each ingester maintains its own consumer group. Each Kafka record is consumed exactly once by each ingester zone and records aren't replayed after being consumed.
```

Comment:
actually let's say "at least once" instead of "exactly once".


> Each Kafka record is consumed at least once by each ingester zone. Records usually are not replayed after being consumed, unless the ingester abruptly shuts down before managing to commit to its consumer group.

(i don't know if this suggestion isn't with too much detail)

Reply:
I think it's okay. Let me know how it sounds to you.